src/components/metadata/styles/metadata-code.scss

```diff
@@ -0,0 +1,184 @@
+@import '../../../styles/variables';
+@import '../../../styles/extends';
+@import '../../../styles/mixins';
+@import './metadata-code-themes.scss';
+
+.kui-theme--light {
+  --color-metadata-code-bg: #{$color-bg-light-4};
+}
+
+.kui-theme--dark {
+  --color-metadata-code-bg: #{$color-bg-dark-3};
+}
+
+.pipeline-metadata-code {
+  @extend %sidebar;
+
+  $margin: 24px;
+  position: absolute;
+  top: -1px; /* Avoids pixel rounding gaps */
+  right: 0;
+  bottom: -1px;
+
+  // On small screens anchor to the left side of the screen
+  left: 0;
+
+  z-index: 2;
+  padding: 0;
+  background: var(--color-metadata-code-bg);
+  transform: translateX(100vw);
+  transition: transform ease 0.5s 0.1s, left ease 0.5s;
+
+  &--visible {
+    transform: translateX(0);
+  }
+
+  @mixin breakpoints($sidebar-width) {
+    // When screen is the minimum usable width
+    @media (min-width: $metadata-sidebar-code-breakpoint) {
+      // Anchor code to the left sidebar
+      left: $sidebar-width + $margin;
+    }
+
+    // When screen is wide enough to fit all sidebars
+    $sidebar-total-width: $sidebar-width + $margin + $metadata-sidebar-code-open +
+      $metadata-sidebar-width-open;
+    @media (min-width: #{$sidebar-total-width}) {
+      // Anchor to right sidebar
+      left: calc(
+        100% - #{$margin + $metadata-sidebar-code-open +
+          $metadata-sidebar-width-open}
+      );
+    }
+  }
+
+  // When sidebar is open
+  &--sidebarVisible {
+    @include breakpoints($sidebar-width-open);
+  }
+
+  // When sidebar is closed
+  &--no-sidebarVisible {
+    @include breakpoints($sidebar-width-closed);
+  }
+
+  @media print {
+    display: none;
+  }
+}
+
+.pipeline-metadata-code__title {
+  flex-grow: 1;
+  margin: 0;
+  margin: 2.1em 36px 1.8em 36px;
+  font-weight: normal;
+  font-size: 1.7em;
+  line-height: 1.6;
+}
+
+.pipeline-metadata-code__code {
+  position: absolute;
+  top: 7.1em;
+  right: $metadata-sidebar-width-open;
+  bottom: 0;
+  left: 0;
+  display: block;
+  overflow: auto;
+  color: var(--color-text);
+  font-size: 1.25em;
+  line-height: 1.8;
+  opacity: 1;
+  transition: opacity 0.4s ease 0.4s;
+
+  .pipeline-metadata-code--no-visible & {
+    opacity: 0;
+    transition: opacity 0.3s ease;
+  }
+
+  * {
+    font-family: Consolas, Monaco, 'Courier New', Courier, monospace;
+  }
+
+  .hljs-keyword {
+    color: var(--color-metadata-code-text);
+  }
+}
+
+.pipeline-metadata-code pre {
+  display: inline-block;
+  margin: 0;
+  padding: 0 36px 36px;
```

Comment:
same as above - should be `3.6em`?

Reply:
As above, I'm not sure that this needs to be in em, because in this case it's not based on font-size but on overall panel width, which is handled in px like most of our layout. There's an argument to be made that literally everything should be done in em, but that's not where we're currently at.

Reply:
Exactly that thanks for clarifying @richardwestenra